src/company_cmd.cpp

```diff
@@ -706,19 +722,13 @@ void OnTick_Companies()
 	}
 
 	if (_next_competitor_start == 0) {
-		/* AI::GetStartNextTime() can return 0. */
-		_next_competitor_start = max(1, AI::GetStartNextTime() * DAY_TICKS);
+		/* If GetStartNextTime is zero, wait at least a day to re-evaluate.
+		 * This avoids checking every tick if an AI can be started. */
+		_next_competitor_start = max(DAY_TICKS, AI::GetStartNextTime() * DAY_TICKS);
```

Comment:
Can I revert this line? The whole point of this PR was so that it could create the correct number of AIs while checking every tick.